src/h5web/vis-packs/core/heatmap/ColorBar.tsx

```diff
@@ -7,24 +7,28 @@ import { getLinearGradient } from './utils';
 import type { ScaleType, Domain } from '../models';
 import type { ColorMap } from './models';
 import { INTERPOLATORS } from './interpolators';
+import { format } from 'd3-format';
+
+const boundFormatter = format('.3~g');
```

Comment:
- `~` trims trailing zeros
- `g` chooses between decimal and exponent notation dynamically